elastio-nat-provision-lambda/cloudformation-lambda.yaml

```diff
@@ -0,0 +1,215 @@
+# yaml-language-server: $schema=https://raw.githubusercontent.com/awslabs/goformation/master/schema/cloudformation.schema.json
+AWSTemplateFormatVersion: '2010-09-09'
+Description: >
+  Provisions a NAT Gateway when Elastio worker instances are running, and de-provisions the gateway when
+  the instances are not running.
+Parameters:
+  DeleteQuiescentPeriodSeconds:
+    Type: Number
+    Default: 300
+    MinValue: 0
+    Description: How long to wait for new EC2 instances to appear before deleting the NAT Gateway
+  CleanupScheduleCron:
+    Type: String
+    Default: '0 1 * * ? *'
+    Description: >
+      A cron expression that defines when and how often to run a cleanup routine.
+      The syntax corresponds to the AWS Scheduler's cron expression syntax:
+      https://docs.aws.amazon.com/scheduler/latest/UserGuide/schedule-types.html#cron-based
+  NatGatewayStackPrefix:
+    Type: String
+    Default: elastio-nat-gateway-
+    MinLength: 1
+    Description: Prefix of the name of the NAT Gateway CFN stack. The name will be <prefix><vpc-id>
+  LambdaMemorySize:
+    Type: Number
+    Default: 512
+    MinValue: 128
+    MaxValue: 10240
+    Description: Amount of memory allocated to the lambda function
+  LambdaTimeout:
+    Type: Number
+    Default: 600
+    MinValue: 10
+    MaxValue: 900
+    Description: Max amount of time the lambda function can run
+
+Resources:
+  lambdaRole:
+    Type: AWS::IAM::Role
+    Properties:
+      AssumeRolePolicyDocument:
+        Version: 2012-10-17
+        Statement:
+          - Effect: Allow
+            Principal:
+              Service: lambda.amazonaws.com
+            Action: sts:AssumeRole
+      ManagedPolicyArns:
+        - !Sub 'arn:aws:iam::aws:policy/service-role/AWSLambdaBasicExecutionRole'
+      Policies:
+        - PolicyName: ElastioNatProvisionPolicy
+          PolicyDocument:
+            Version: 2012-10-17
+            Statement:
+              - Effect: Allow
+                Action:
```

Comment:
Am I missing the grants for CloudWatch logs?  It's important that we have Cloudwatch logs for troubleshooting.

I would also like to see the Cloudwatch log group provisioned as part of this CFN, so that we can set a reasonably short retention period on it to avoid cost runaways.

Reply:
Why would the Lambda name include the VPC ID and AZ?  The Lambda will be listening for any EventBridge events about EC instance regardless of VPC or AZ.

Reply:
right, the lambda name will have a static name (`elastio-nat-gateway-provision`)